src/gatsby-node.ts

```diff
@@ -16,17 +16,37 @@ interface ServerlessRoutingRule {
 const getRules = (pluginOptions: PluginOptions, routes: GatsbyRedirect[]): RoutingRules => (
     routes.map(route => ({
         Condition: {
-            KeyPrefixEquals: withoutLeadingSlash(route.fromPath),
-            HttpErrorCodeReturnedEquals: '404',
```

Comment:
the following assumes that 404 is returned but depending on your S3 permissions, you may also get 403. removed it cause one of the two will suffice for the redirect